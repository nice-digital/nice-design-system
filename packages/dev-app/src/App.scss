@import '~@nice-digital/design-system/scss/nice-design-system';

main {
  background: white;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.3);
  position: relative;
}

.grid-example-item {
  background: $nds-colour-nice-grey-3;
  border: 0.25em solid $nds-colour-nice-grey-1;
  margin-bottom: 0.5em;
  padding: 0.5em;
  text-align: center;
}
