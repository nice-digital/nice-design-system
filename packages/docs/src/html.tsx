import React from "react";
import PropTypes from "prop-types";

export default function HTML(props: any): React.ReactElement {
	return (
		<html {...props.htmlAttributes}>
			<head>
				<meta charSet="utf-8" />
				<meta httpEquiv="x-ua-compatible" content="ie=edge" />
				<link
					href="https://fonts.googleapis.com/css?family=Lato:200normal,200italic,300normal,300italic,400normal,400italic,700normal,700italic,900normal,900italic"
					rel="stylesheet"
					type="text/css"
				/>
				<meta
					name="viewport"
					content="width=device-width, initial-scale=1, shrink-to-fit=no"
				/>
				{props.headComponents}
			</head>
			<body {...props.bodyAttributes}>
				{props.preBodyComponents}
				<div
					key={"body"}
					id="___gatsby"
					dangerouslySetInnerHTML={{ __html: props.body }}
				/>
				{props.postBodyComponents}
			</body>
		</html>
	);
}

HTML.propTypes = {
	htmlAttributes: PropTypes.object,
	headComponents: PropTypes.array,
	bodyAttributes: PropTypes.object,
	preBodyComponents: PropTypes.array,
	body: PropTypes.string,
	postBodyComponents: PropTypes.array
};
