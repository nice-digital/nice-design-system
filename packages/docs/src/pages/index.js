import React from "react";
import { Hero } from "@nice-digital/nds-hero";

const IndexPage = () => (
	<>
		<Hero>Hello</Hero>
	</>
);

export default IndexPage;
