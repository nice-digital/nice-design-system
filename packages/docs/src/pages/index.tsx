import React from "react";
import { Hero } from "@nice-digital/nds-hero";
import { Grid, GridItem } from "@nice-digital/nds-grid";
// @ts-ignore
import { PhaseBanner } from "@nice-digital/nds-phase-banner";
import { graphql, Link } from "gatsby";
import Seo from "../components/partials/Seo";
import Wrapper from "../components/layouts/Wrapper";
import { Breadcrumbs, Breadcrumb } from "@nice-digital/nds-breadcrumbs";

export const query = graphql`
	{
		allSitePage {
			edges {
				node {
					path
					id
				}
			}
		}
		site {
			siteMetadata {
				homeLabel
			}
		}
	}
`;

export default function IndexPage(props: any) {
	return (
		<Wrapper>
			<Seo title="NICE Design System" />
			<Hero
				title="NICE Design System"
				intro="Your source for quickly creating consistent on-brand NICE digital services"
				header={
					<>
						<PhaseBanner phase="alpha">
							NICE Design System is in development. This means it is not feature
							complete and there may be issues. Find any? Please,{" "}
							<a
								href="https://github.com/nice-digital/nice-design-system/issues"
								target="_blank"
								rel="noreferrer noopener"
							>
								let us know
							</a>
							!
						</PhaseBanner>
						<Breadcrumbs>
							<Breadcrumb to="/" elementType={Link}>
								{props.data.site.siteMetadata.homeLabel}
							</Breadcrumb>
						</Breadcrumbs>
					</>
				}
				actions={
					<>
						<Link to="/about/getting-started" className="btn btn--cta">
							Get started
						</Link>
						<a
							href="https://github.com/nice-digital/nice-design-system"
							className="btn"
							target="_blank"
							rel="noreferrer noopener"
						>
							View it on Github
						</a>
					</>
				}
			>
				<h2 className="h4 mt--0-md">Quick links</h2>
				<ul className="list--unstyled">
					<li>
						<Link to="/foundations">Foundations</Link>
					</li>
					<li>
						<Link to="/components">Components</Link>
					</li>
					<li>
						<Link to="/technical">Technical</Link>
					</li>
					<li>
						<Link to="/about">About the Design System</Link>
					</li>
				</ul>
			</Hero>
			<Grid>
				<GridItem cols={12}>
					<ul>
						{props.data.allSitePage.edges.map((edge: any) => {
							let { id, path } = edge.node;
							return (
								<li key={id}>
									<Link to={path}>{path}</Link>
								</li>
							);
						})}
					</ul>
				</GridItem>
			</Grid>
		</Wrapper>
	);
}
