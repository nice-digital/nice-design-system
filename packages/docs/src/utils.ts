import { ResponseObjectType } from "./components/partials/navigation/Navigation";

/**
 * Make the first character of a string uppercase
 *
 * @return {string} A capitalised string or null
 * @param string
 */
export function capitalise(string: string): string | null {
	return string ? string.charAt(0).toUpperCase() + string.slice(1) : null;
}

/**
 * Does a query item's slug start with the string provided?
 *
 * @return { boolean } Whether or not the string provided matches the start
 * @param item A graphql response object
 * @param this The slug to match against
 */
export function slugMatches(item: ResponseObjectType) {
	const parentSlug: string = this;
	return item.slug.indexOf(parentSlug) === 0;
}

/**
 * Format a query item into one suitable for navigation
 *
 * @param item A graphql response object
 * @param this The current item's ID to assess whether current
 * @return object
 */
export function formatList(item: ResponseObjectType) {
	const currentId: string = this;
	return {
		label: item.frontmatter.navigationLabel || item.frontmatter.title,
		destination: item.slug,
		isCurrent: item.id === currentId,
		order: item.frontmatter.order || 999
	};
}

/**
 * Is an item's slug a sibling or direct child of the slug provided
 * @param item { object } A graphql response object
 * @param this { number } The current slug's length
 * @return { boolean } Is a sibling or a direct child
 */
export function isSiblingOrDirectChild(item: ResponseObjectType) {
	const currentSlugLength: number = this;
	const itemSlugLength = item.slug.split("/").filter(i => i).length;
	return (
		itemSlugLength === currentSlugLength || // we're at the same "level"
		itemSlugLength === currentSlugLength + 1 // or at one "level" below
	);
}

/**
 * Is an item's slug a direct child of the slug provided
 * @param item { object } A graphql response object
 * @param this { number } The current slug's length
 * @return { boolean } Is a direct child
 */
export function isDirectChild(item: ResponseObjectType) {
	const currentSlugLength: number = this;
	const itemSlugLength = item.slug.split("/").filter(i => i).length;
	return (
		itemSlugLength === currentSlugLength + 1 // or at one "level" below
	);
}
