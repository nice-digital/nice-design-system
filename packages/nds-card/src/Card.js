import React from "react";
import PropTypes from "prop-types";
import "../scss/card.scss";

const CardHeader = props => {
	const { headingText, elementType: HeadingTag = "p", link } = props;
	let linkProps = {};
	if (link) {
		const { elementType: LinkTag = "a", destination } = link;
		if (LinkTag === "a") {
			linkProps.href = destination;
		} else {
			linkProps.to = destination;
		}
		return (
			<HeadingTag className="card__heading">
				<LinkTag {...linkProps}>{headingText}</LinkTag>
			</HeadingTag>
		);
	} else {
		return <HeadingTag className="card__heading">{headingText}</HeadingTag>;
	}
};

export const Card = props => {
	const { metadata } = props;
	return (
		<article className="card">
			<header className="card__header">
				<CardHeader {...props} />
			</header>
			{metadata && metadata.length && (
				<dl className="card__metadata">
					{metadata.map((item, idx) => {
						if (!item.value) return null;
						return (
							<div key={`item${idx}`} className="card__metadatum">
								{item.label && (
									<dt className="visually-hidden">{item.label}</dt>
								)}
								<dd>{item.value}</dd>
							</div>
						);
					})}
				</dl>
			)}
		</article>
	);
};

Card.propTypes = {
	headingText: PropTypes.node.isRequired,
	elementType: PropTypes.elementType,
	link: PropTypes.shape({
		destination: PropTypes.node,
		elementType: PropTypes.elementType
	}),
	metadata: PropTypes.arrayOf(
		PropTypes.shape({
			label: PropTypes.node,
			value: PropTypes.node.isRequired
		})
	)
};
