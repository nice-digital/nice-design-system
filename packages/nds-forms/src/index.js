import Textarea from "./Textarea";
import Input from "./Input";

export { Textarea, Input };
