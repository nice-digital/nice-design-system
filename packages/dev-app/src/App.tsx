import React, { Component } from "react";
// import Faker from "faker";
import { Router, Route, Link } from "react-router-dom";
import { createBrowserHistory } from "history";
import { FormGroupView } from "./views/FormGroupView";
import { PageHeaderView } from "./views/PageHeaderView";
import { GridView } from "./views/GridView";
import { AlertView } from "./views/AlertView";
import { StackedNavView } from "./views/StackedNavView";
import { CardView } from "./views/CardView";
import { ButtonView } from "./views/ButtonView";
import { HeroView } from "./views/HeroView";
import { TabsView } from "./views/TabsView";
import { TagsView } from "./views/TagsView";
import { MaintainRatioView } from "./views/MaintainRatioView";
import { PrintView } from "./views/PrintView";
import { BreadcrumbsView } from "./views/BreadcrumbsView";
import { FiltersView } from "./views/FiltersView";
import { DeprecatedFormsView } from "./views/DeprecatedFormsView";
import { SimplePaginationView } from "./views/SimplePaginationView";
import "./App.scss";

const browserHistory = createBrowserHistory();

class App extends Component {
	render() {
		return (
			<Router history={browserHistory}>
				<ul className="container hide-print list list--unstyled">
					<li>
						<Link to="/formgroup">Form Group</Link>
					</li>
					<li>
						<Link to="/pageheader">Page Header</Link>
					</li>
					<li>
						<Link to="/stackednav">Stacked Nav</Link>
					</li>
					<li>
						<Link to="/alert">Alert</Link>
					</li>
					<li>
						<Link to="/grid">Grid</Link>
					</li>
					<li>
						<Link to="/card">Card</Link>
					</li>
					<li>
						<Link to="/button">Button</Link>
					</li>
					<li>
						<Link to="/hero">Hero</Link>
					</li>
					<li>
						<Link to="/tabs">Tabs</Link>
					</li>
					<li>
						<Link to="/tags">Tags</Link>
					</li>
					<li>
						<Link to="/maintain-ratio">Maintain Ratio</Link>
					</li>
					<li>
						<Link to="/print">Print</Link>
					</li>
					<li>
						<Link to="/breadcrumbs">Breadcrumbs</Link>
					</li>
					<li>
						<Link to="/filters">Filters</Link>
					</li>
					<li>
						<Link to="/dep-forms">Deprecated Forms</Link>
					</li>
					<li>
						<Link to="/simple-pagination">Simple pagination</Link>
					</li>
				</ul>
				<main className="container">
					<Route path="/formgroup" component={FormGroupView} />
					<Route path="/pageheader" component={PageHeaderView} />
					<Route path="/stackednav" component={StackedNavView} />
					<Route path="/alert" component={AlertView} />
					<Route path="/hero" component={HeroView} />
					<Route path="/grid" component={GridView} />
					<Route path="/card" component={CardView} />
					<Route path="/button" component={ButtonView} />
					<Route path="/tabs" component={TabsView} />
					<Route path="/tags" component={TagsView} />
					<Route path="/maintain-ratio" component={MaintainRatioView} />
					<Route path="/print" component={PrintView} />
					<Route path="/breadcrumbs" component={BreadcrumbsView} />
					<Route path="/filters" component={FiltersView} />
					<Route path="/dep-forms" component={DeprecatedFormsView} />
					<Route path="/simple-pagination" component={SimplePaginationView} />
				</main>
			</Router>
		);
	}
}

export default App;
