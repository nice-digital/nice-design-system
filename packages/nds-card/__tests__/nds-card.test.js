import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import { Card } from "../src/Card";
import { Link, MemoryRouter } from "react-router-dom";

const headingProps = {
	headingText: "Card heading text",
	elementType: "h1",
	link: {
		destination: "/about",
		elementType: "a"
	}
};
const metadataProps = [
	{
		value: "john.smith@email.com",
		label: "Email address"
	},
	{
		value: "sarah.smith@email.com",
		label: "Email address"
	}
];

const props = {
	...headingProps,
	metadata: metadataProps
};

describe("Card", () => {
	it("should render without crashing", () => {
		const wrapper = shallow(<Card {...props}>Body</Card>);
		expect(wrapper).toHaveLength(1);
	});

	it("should match snapshot with some default attributes", () => {
		const wrapper = shallow(<Card {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("should render each metadata item with a value", () => {
		const wrapper = shallow(<Card {...props} />);
		expect(wrapper.find(".card__metadatum")).toHaveLength(2);
	});

	it("should not render an a metadata item without a value", () => {
		const localMetadataProps = [...metadataProps];
		localMetadataProps[0].value = "";
		const wrapper = shallow(
			<Card {...headingProps} metadata={localMetadataProps} />
		);
		expect(wrapper.find(".card__metadatum")).toHaveLength(1);
	});

	it("should render an href attribute if the elementType is an anchor", () => {
		const wrapper = mount(<Card {...headingProps} />);
		const anchor = wrapper.find("a");
		expect(anchor.props()["href"]).toEqual("/about");
	});

	it("should render a 'to' attribute if the elementType is anything other than an anchor", () => {
		const localHeadingProps = Object.assign({}, headingProps);
		localHeadingProps.link.elementType = Link;
		const wrapper = mount(
			<MemoryRouter>
				<Card {...localHeadingProps} />
			</MemoryRouter>
		);
		const anchor = wrapper.find(Link);
		expect(anchor.props()["to"]).toEqual("/about");
	});
});
