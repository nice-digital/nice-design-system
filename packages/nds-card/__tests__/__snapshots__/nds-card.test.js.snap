// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot with some default attributes 1`] = `
<article
  className="card"
>
  <header
    className="card__header"
  >
    <CardHeader
      headingElementType="h1"
      headingText="Card heading text"
      link={
        Object {
          "destination": "/about",
          "elementType": "a",
        }
      }
    />
  </header>
  <dl
    className="card__metadata"
  >
    <div
      className="card__metadatum"
      key="item0"
    >
      <dt
        className="visually-hidden"
      >
        Email address
      </dt>
      <dd>
        john.smith@email.com
      </dd>
    </div>
    <div
      className="card__metadatum"
      key="item1"
    >
      <dt
        className="visually-hidden"
      >
        Email address
      </dt>
      <dd>
        sarah.smith@email.com
      </dd>
    </div>
  </dl>
</article>
`;
